src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -1405,9 +1405,7 @@ internal static bool IsAttributeDefined(RuntimeModule decoratedModule, int decor
                     {
                         // Metadata is always written in little-endian format. Must account for this on
                         // big-endian platforms.
-                        const int CustomAttributeVersion = BitConverter.IsLittleEndian ? 0x0001 : 0x0100;
-
-                        if (Marshal.ReadInt16(blobStart) != CustomAttributeVersion)
+                        if (Marshal.ReadInt16(blobStart) != (BitConverter.IsLittleEndian ? 0x0001 : 0x0100))
```

Comment:
It looks like `blobStart` is a pointer. Might be worth a bug investigating using `Span<>` and `BinaryPrimitives.Read*LittleEndian`?